packages/peregrine/lib/targets/peregrine-intercept.js

```diff
@@ -0,0 +1,68 @@
+/**
+ * @module Peregrine/Targets
+ */
+const path = require('path');
+
+/**
+ *
+ * @class TalonWrapperConfig
+ * @hideconstructor
+ */
+function TalonWrapperConfig(addTransforms) {
+    const wrappable = (talonFile, exportName) => ({
+        wrapWith: wrapperModule =>
+            addTransforms({
+                type: 'source',
+                fileToTransform: path.join('./lib/talons/', talonFile),
+                transformModule:
+                    '@magento/pwa-buildpack/lib/WebpackTools/loaders/wrap-esm-loader',
+                options: {
+                    wrapperModule,
+                    exportName
+                }
+            })
+    });
+    return {
+        /**
+         * @memberof TalonWrapperConfig
+         *
+         */
+
+        ProductFullDetail: {
+            useProductFullDetail: wrappable(
+                'ProductFullDetail/useProductFullDetail',
+                'useProductFullDetail'
+            )
+        },
+        App: {
+            useApp: wrappable('App/useApp', 'useApp')
```

Comment:
This feels a bit too verbose. Why do we need so much repetition here around the component and talon names?

Reply:
The second argument is the name of the export to wrap. All of the talon files export their main talon functions as a named export, so that looks like our convention, but are we sticking to it?

If so, we can maybe automate some of what this class does, since its implementation is internal to Peregrine.

Reply:
I'm not sure why we export named talons vs just have them as the default but it may be too late to make such a sweeping change anyways. I'd prefer we not have to provide the export name. I had assumed we would just automate intercepts around all talons by default.

@jimbo may have an opinion about this.

Reply:
@sirugh Let's definitely wait for @jimbo to have input, but I think this can be a followup too. The point of the `TalonInterceptorConfig` is to be a facade for _however_ we are organizing Peregrine talons. If we change it, we can modify the implementation.

Reply:
> This feels a bit too verbose. Why do we need so much repetition here around the component and talon names?

Yes, it's verbose and repetitive because it's explicit, which is a Good Thing here. We certainly wouldn't just tap into every export (`*`), and I don't think tapping into the default export would save us a whole lot of verbosity. I don't really share your concern here, I guess.

> I'm not sure why we export named talons vs just have them as the default but it may be too late to make such a sweeping change anyways. I'd prefer we not have to provide the export name. I had assumed we would just automate intercepts around all talons by default.

When you and I wrote these talons, React hooks were new. I elected to make them named exports because I was uncertain that our developer users would grasp the *magical requirement* that hooks must be named `useFoo` in order for React to handle them correctly, so as a precaution, I wanted to discourage renaming talons on import.

As it turns out, we haven't really had an issue. The React community at large has adopted the `useFoo` convention without any problems or controversy (much less controversy than JSX, even), and our contributors and downstream users are using our hooks (and writing their own!) without any problems. So we could switch from named exports to default exports for talons now, but I don't think that amount of churn would be very justifiable.

Reply:
That was a very bright idea in the first place. I thought it was general doubt about default exports, but the enforcing of a naming convention was very smart.

We can reduce complexity by switching to default exports. I don't think much churn would be required, because you can just put `export useFoo as deefault` at the bottom of every file, thus preserving the old and new APIs.

Reply:
> We can reduce complexity by switching to default exports. I don't think much churn would be required, because you can just put `export useFoo as default` at the bottom of every file, thus preserving the old and new APIs.
>
> Wanna schedule that, and mark the verbosity in this file as tech debt? It's up to y'all.

Yes, actually. If we were to add a default export to the talons, and deprecate but not remove the named export, it wouldn't even require us to bump the major version.

```es6
export const useFoo = () => {}

export default useFoo
```

Reply:
OK, I'll push that as a todo.